esphome/components/max31865/max31865.cpp

```diff
@@ -0,0 +1,171 @@
+#define MAX31865_CFG_MODE (0x40)  // 1 - Auto conversion
+#define MAX31865_CFG_BIAS (0x80)  // 1 - Vbias Enabled
+#define MAX31865_CFG_WIRE_EVEN(x) x ? 0 : MAX31865_CFG_WIRE
+
+#define MAX31865_RTD_L_FLT (0x01)  // Indicates Fault
+
+#define MAX31865_FLT_OVUV (0x04)
+#define MAX31865_FLT_RTNO (0x08)
+#define MAX31865_FLT_RFNO (0x10)
+#define MAX31865_FLT_RFNS (0x20)
+#define MAX31865_FLT_TSHL (0x40)
+#define MAX31865_FLT_TSHH (0x80)
+
+#define RTD_A 3.9083e-3
+#define RTD_B -5.775e-7
+
+namespace esphome {
+namespace max31865 {
+
+static const char *TAG = "max31865";
+
+float MAX31865Sensor::temperature(unsigned short adc, float rtd_nominal, float ref_resistor) {
+  // https://github.com/adafruit/Adafruit_MAX31865/blob/master/Adafruit_MAX31865.cpp
+  // http://www.analog.com/media/en/technical-documentation/application-notes/AN709_0.pdf
+
+  float z1, z2, z3, z4, r_t, temp;
+
+  r_t = adc;
+  r_t /= 32768;
+  r_t *= ref_resistor;
+
+  ESP_LOGD(TAG, "'%s': Resistance: %.3f", this->name_.c_str(), r_t);
+
+  z1 = -RTD_A;
+  z2 = RTD_A * RTD_A - (4 * RTD_B);
+  z3 = (4 * RTD_B) / rtd_nominal;
+  z4 = 2 * RTD_B;
+
+  temp = z2 + (z3 * r_t);
+  temp = (std::sqrt(temp) + z1) / z4;
+
+  if (temp >= 0)
+    return temp;
+
+  // ugh.
+  r_t /= rtd_nominal;
+  r_t *= 100;  // normalize to 100 ohm
+
+  float rpoly = r_t;
+
+  temp = -242.02;
+  temp += 2.2228 * rpoly;
+  rpoly *= r_t;  // square
+  temp += 2.5859e-3 * rpoly;
+  rpoly *= r_t;  // ^3
+  temp -= 4.8260e-6 * rpoly;
+  rpoly *= r_t;  // ^4
+  temp -= 2.8183e-8 * rpoly;
+  rpoly *= r_t;  // ^5
+  temp += 1.5243e-10 * rpoly;
+
+  return temp;
+}
+
+void MAX31865Sensor::write_config_() {
+  this->enable();
+  delay(1);
+  this->write_byte(MAX31865_CFG | MAX31865_WRITE);
+  this->write_byte(MAX31865_CFG_FLTC | MAX31865_CFG_WIRE_EVEN(this->even_pins_) | MAX31865_CFG_MODE |
+                   MAX31865_CFG_BIAS);  // TODO: Allow user to specify configuration
+  this->disable();
+}
+
+void MAX31865Sensor::update() {
+  // Conversion time typ: 52ms (60Hz mode), max: 220ms
+  auto f = std::bind(&MAX31865Sensor::read_data_, this);
+  this->set_timeout("value", 220, f);
+}
+
+void MAX31865Sensor::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up MAX31865Sensor '%s'...", this->name_.c_str());
+  this->spi_setup();
+
+  // Configuire MAX31865 device
+  this->write_config_();
+  // TODO: Run fault-detection cycle
+}
+
+void MAX31865Sensor::dump_config() {
+  LOG_SENSOR("", "MAX31865", this);
+  LOG_PIN("  CS Pin: ", this->cs_);
+  LOG_UPDATE_INTERVAL(this);
+}
+
+float MAX31865Sensor::get_setup_priority() const { return setup_priority::DATA; }
+
+void MAX31865Sensor::read_data_() {
+  this->enable();
+  delay(1);
+  this->write_byte(MAX31865_RTD_H);
+  uint8_t data[2];
+  this->read_array(data, 2);
+  this->disable();
+
+  uint16_t val = (data[1] | (data[0] << 8)) >> 1;
+
+  ESP_LOGD(TAG, "'%s': Read ADC of 0x%04X", this->name_.c_str(), val);
```

Comment:
```suggestion
  ESP_LOGVV(TAG, "'%s': Read ADC of 0x%04X", this->name_.c_str(), val);
```

This is a verbose message (all raw communications)